core/embed/rust/src/ui/layout_delizia/component/trade_screen.rs

```diff
@@ -0,0 +1,73 @@
+use crate::{
+    strutil::TString,
+    ui::{
+        component::{paginated::PaginateFull, Bar, Component, Event, EventCtx, Label, Never},
+        geometry::Rect,
+        shape::Renderer,
+        util::Pager,
+    },
+};
+
+use super::super::theme;
+
+pub struct TradeScreen {
+    sell_amount: Label<'static>,
+    line: Bar,
+    buy_amount: Label<'static>,
+}
+
+impl TradeScreen {
+    pub fn new(sell_amount: TString<'static>, buy_amount: TString<'static>) -> Self {
+        Self {
+            sell_amount: Label::left_aligned(sell_amount, theme::TEXT_WARNING).top_aligned(),
+            line: Bar::new(theme::GREY_EXTRA_DARK, theme::BG, 1),
+            buy_amount: Label::left_aligned(buy_amount, theme::TEXT_MAIN_GREEN_LIME)
+                .bottom_aligned(),
+        }
+    }
+}
+
+impl PaginateFull for TradeScreen {
+    fn pager(&self) -> Pager {
+        Pager::new(1).with_current(0)
+    }
+
+    fn change_page(&mut self, _active_page: u16) {
+        unimplemented!()
+    }
+}
+
+impl Component for TradeScreen {
+    type Msg = Never;
+
+    fn place(&mut self, bounds: Rect) -> Rect {
+        let (top, bottom) = bounds.split_top(bounds.height() / 2);
```

Comment:
Here, the grid might be a bit easier to read
```suggestion
        let grid = Grid::new(bounds, 4, 1)
        self.sell_amount.place(grid.row_col(1, 0));
        self.buy_amount.place(grid.row_col(2, 0));
```